apps/patient/src/views/Pharmacy.tsx

```diff
@@ -113,127 +279,152 @@ export const Pharmacy = () => {
-    const data = await geocoder.geocode({ address });
-    if (data?.results) {
-      return {
-        loc: data.results[0].formatted_address,
-        lat: data.results[0].geometry.location.lat(),
-        lng: data.results[0].geometry.location.lng()
-      };
+  const initialize = async () => {
+    setLoadingPharmacies(true);
+
+    // Perform geocode to get lat/lng
+    let locationData: { address: string; lat: number; lng: number };
+    try {
+      locationData = await geocode(location);
+    } catch (error) {
+      console.error('Geocoding error:', error);
     }
-  };
 
-  const fetchPharmacies = async () => {
-    setLoadingMore(true);
-
-    // On initializing, we won't have lat/lng
-    let loc: string = location;
-    let lat: number = latitude;
-    let lng: number = longitude;
-    if (!lat || !lng) {
-      const geo = await geocode(location);
-      if (!geo) return;
-
-      loc = geo.loc;
-      lat = geo.lat;
-      lng = geo.lng;
-
-      // Save location data for show more pharmacies
-      setLocation(loc);
-      setLatitude(lat);
-      setLongitude(lng);
+    const { address, lat, lng } = locationData;
+    if (!address || !lat || !lng) {
+      setLoadingPharmacies(false);
+      return;
     }
 
-    // Get pharmacies from our internal list
-    const locationData = {
-      latitude: lat,
-      longitude: lng,
-      radius: 25
-    };
-    const limit = 3;
-    const offset = pharmacyOptions.length;
+    setLocation(address);
+    setLatitude(lat);
+    setLongitude(lng);
+
+    // Get pharmacies from photon db
+    const pharmaciesResult = await getPharmacies(
+      {
+        latitude: lat,
+        longitude: lng,
+        radius: 25
+      },
+      30, // Set high to ensure indie's are found. Request time increase is minimal.
+      0,
+      token
+    );
+    if (pharmaciesResult === 'No pharmacies found near location') {
+      setLoadingPharmacies(false);
+      toast({
+        title: pharmaciesResult,
+        position: 'top',
+        status: 'warning',
+        duration: 5000,
+        isClosable: true
+      });
+      return;
+    }
+    if (!pharmaciesResult) {
+      setLoadingPharmacies(false);
+      return;
+    }
 
-    graphQLClient.setHeader('x-photon-auth', token);
+    // Save in case we fetched more than we initially showed
+    setFetchedPharmacies(pharmaciesResult);
 
-    let pharmaciesResults: any;
-    try {
-      pharmaciesResults = await graphQLClient.request(GET_PHARMACIES, {
-        location: locationData,
-        limit,
-        offset
-      });
-    } catch (error) {
-      console.error(JSON.stringify(error, undefined, 2));
-      console.log(error);
-      if (error?.response.errors[0].message === 'No pharmacies found near location') {
-        setShowingAllPharmacies(true);
+    // Prepare pharmacies to show
+    let newPharmacies: PharmacyType[] = pharmaciesResult;
+
+    // Feature pharmacies
+    newPharmacies = sortIndiePharmaciesFirst(
```

Comment:
will this happen after a patient clicks "load more"? wondering if it's weird that there are indie pharmacies at the top of every click load more

Reply:
Ah looks like this is only called once, and enrichPharmacies is called for subsequent calls. Will move this comment down there

Reply:
nope, just on initialize(), which happens on initial load or after location change